### PR TITLE
People: Only show invite form for user with promote_users capability

### PR DESCRIPTION
**Pull request by ebinnion**
Since inviting users requires `promote_users` capability, we should only show the form for users with that capability (administrators). This PR makes use of the `EmptyContent` component and our Drake image to match what we show when attempting to go to `/people/team/$site` without `list_users` capability.

To test:
- Checkout `update/invites-fix-no-permissions` branch
- Log into WordPress.com
- Go to `/people/new/$site` where your user is an admin
- Do you see the form? Good.
- Go to `/people/new/$site` where you are an editor or lower role
- Do you see our good friend Drake? Good
- Run `localStorage.clear()` in console and refresh
- Do you see the invite form until sites are fetched? Good

cc @lezama for code review and @rickybanister for design review.

After:

screen shot 2016-03-17 at 11 55 41 am

### CODE REVIEW

**Comment by rickybanister** [comment]
How about "Oops, only administrators can invite other people"

**Comment by ebinnion** [comment]
Updated. Here's a new screenshot:

screen shot 2016-03-17 at 12 18 23 pm

**Comment by rickybanister** [comment]
Can you throw a `&nbsp;` between other and people in case others have tiny screens?

**Comment by ebinnion** [comment]
Updated:

screen shot 2016-03-17 at 12 36 23 pm

**Comment by rickybanister** [comment]
Ship :ship: :truck:

File: client/my-sites/people/invite-people/index.jsx [hunk 1]
```diff
@@ -270,6 +271,18 @@ const InvitePeople = React.createClass( {
 	},
 
 	render() {
+		if ( this.props.site && ! get( this.props, 'site.capabilities.promote_users' ) ) {
+			return (
+				<Main>
+					<SidebarNavigation />
+					<EmptyContent
+						title={ this.translate( 'Oops, only administrators can invite other\u00a0people' ) }
```
Comment (lezama) [inline client/my-sites/people/invite-people/index.jsx]: This is really not i18n friendly :smile:

@rickybanister I think we should find a more generic solution for long phrases.

Comment (rickybanister) [inline client/my-sites/people/invite-people/index.jsx]: Yeah, I'm not sure–we bump into this all the time, I don't really think words are targetable with css...

**Comment by ebinnion** [comment]
My understanding from the above comments was that we should optimize for the easier to translate string rather than a trailing single word at some breakpoints. So, I went ahead and removed the non-breaking space unicode character from the translation.

**Comment by rickybanister** [comment]
I suppose that's fine for now. I'd be a good thing to pursue a solution for all of calypso some day.

File: client/my-sites/people/invite-people/index.jsx [hunk 2]
```diff
@@ -270,6 +271,18 @@ const InvitePeople = React.createClass( {
 	},
 
 	render() {
+		if ( this.props.site && ! get( this.props, 'site.capabilities.promote_users' ) ) {
```
Comment (enejb) [inline client/my-sites/people/invite-people/index.jsx]: You should use the userCan utility function here. https://github.com/Automattic/wp-calypso/blob/master/client/lib/site/utils.js#L7

File: client/my-sites/people/invite-people/index.jsx [hunk 3]
```diff
@@ -270,6 +271,18 @@ const InvitePeople = React.createClass( {
...
+			return (
```
Comment (enejb) [inline client/my-sites/people/invite-people/index.jsx]: It might be nice to wrap this up in its own function so that render is not that long.

**Comment by enejb** [comment]
Beside the minor thing that I found. I have tested this and it worked as expected. Feel free to merge this branch after fixing the minor issues. :shipit:

File: client/my-sites/people/invite-people/index.jsx (hunk 3 above)
Comment (ebinnion) [inline client/my-sites/people/invite-people/index.jsx]: I'm not sure that this is needed quite yet. Currently there's only the one conditional that determine whether we render the form or the error. Also, if the goal is to make `render` small, then we should probably factor out the form into something like `renderInviteForm()`, which would add roughly 70 lines of code to this PR.

I'm not completely against that, but if you think that we should, I'd rather do that in another PR.

**Pull request merged**